README.md

```diff
@@ -1,127 +1,130 @@
+*Flaxs* was created initially as a fork of [McFly](https://github.com/kenwheeler/mcfly).  
+Thanks to [kenwheeler](https://github.com/kenwheeler) for letting this project to happen.
 
-## Download
 
-McFly can be downloaded from:
+## Examples
 
-[http://kenwheeler.github.io/mcfly/McFly.js](http://kenwheeler.github.io/mcfly/McFly.js)
+You can review in detail our [examples guideline page](./tree/master/examples/README.md):
+
+## Demo Page
+
+Right now, we don't have a demo page, nonetheless we have an examples folder which will allow us to explain in detail the potential of Flaxs.
 
 ## Dispatcher
 
-McFly uses Facebook Flux's dispatcher. When McFly is instantiated, a single dispatcher instance is created and can be accessed like shown below:
+Flaxs uses Facebook Flux's dispatcher. When Flaxs is instantiated, a single dispatcher instance is created and can be accessed like shown below:
 
 ```javascript
-var mcFly = new McFly();
+import { flaxs } from 'flaxs';
 
-return mcFly.dispatcher;
+return flaxs.dispatcher;
 ```
-In fact, all created Actions & Stores are also stored on the McFly object as `actions` and `stores` respectively.
+In fact, all created Actions & Stores are also stored on the Flaxs object as `actions` and `stores` respectively.
 
 ## Stores
 
-McFly has a **createStore** helper method that creates an instance of a Store. Store instances have been merged with EventEmitter and come with **emitChange**, **addChangeListener** and **removeChangeListener** methods built in.
+**Stores in Flaxs are used to preserve compatibility with McFly approach, but are not recommended to use for new projects.**
+
+Since Flaxs brings the `reducer` functions, we don't need different places to store the state of the application.  Instead we use the `MasterStore` explained in this document forward.
 
 When a store is created, its methods parameter specifies what public methods should be added to the Store object. Every store is automatically registered with the Dispatcher and the `dispatcherID` is stored on the Store object itself, for use in `waitFor` methods.
 
-Creating a store with McFly looks like this:
+Creating a store with Flaxs using pure functions, you do this:
 
 ```javascript
 var _todos = [];
 
-function addTodo(text) {
-  _todos.push(text);
-}
+const addTodo = (todos, text) => ([
+  ...todos,
+  text,
+]);
 
-var TodoStore = mcFly.createStore({
+const TodoStore = flaxs.createStore({
 
-getTodos: function() {
-  return _todos;
-}
+  getTodos: function() {
+    return _todos;
+  }
 
-}, function(payload){
-  var needsUpdate = false;
+}, function({ actionType, params }){
+  let newTodos;
 
-  switch(payload.actionType) {
+  switch(actionType) {
   case 'ADD_TODO':
-    addTodo(payload.text);
-    needsUpdate = true;
+    if (!_todos.includes(params.text)) {
+      newTodos = addTodo(_todos, params.text);
+    }
     break;
   }
 
-  if (needsUpdate) {
+  if (newTodos !== undefined && newTodos !== _todos) {
```

Comment:
Well let's say we add a `emitChangeIfStoreChanged()` method in the parent class.

This wouldn't break the existing code, nor will it break stores that already implement the "old-style" code.
It could be something that is available for any new store implementation, and the old way can be deprecated as we go.
Correct me if I'm wrong.